kratos/sources/model_part.cpp

```diff
@@ -1594,6 +1594,227 @@ void ModelPart::RemoveConditionsFromAllLevels(Flags IdentifierFlag)
 ///@name Geometry Container
 ///@{
 
+ModelPart::GeometryType::Pointer ModelPart::CreateNewGeometry(
+    std::string GeometryTypeName,
+    std::vector<IndexType> GeometryNodeIds
+    )
+{
+    if (IsSubModelPart()) {
+        GeometryType::Pointer p_new_geometry = mpParentModelPart->CreateNewGeometry(GeometryTypeName, GeometryNodeIds);
+        this->AddGeometry(p_new_geometry);
+        return p_new_geometry;
+    }
+
+    GeometryType::PointsArrayType p_geometry_nodes;
+    for (IndexType i = 0; i < GeometryNodeIds.size(); ++i) {
+        p_geometry_nodes.push_back(pGetNode(GeometryNodeIds[i]));
+    }
+
+    return CreateNewGeometry(GeometryTypeName, p_geometry_nodes);
+}
+
+ModelPart::GeometryType::Pointer ModelPart::CreateNewGeometry(
+    std::string GeometryTypeName,
+    GeometryType::PointsArrayType pGeometryNodes
+    )
+{
+    KRATOS_TRY
+
+        if (IsSubModelPart()) {
+            GeometryType::Pointer p_new_geometry = mpParentModelPart->CreateNewGeometry(GeometryTypeName, pGeometryNodes);
```

Comment:
does this also work if there are multiple (several!) levels of submodelparts?

Reply:
Should be, it is the same strategy followed in the elements creation

Reply:
I'll have to check

Reply:
oh, it's recursive, allright then